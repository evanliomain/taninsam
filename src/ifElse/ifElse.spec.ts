import { ifElse } from './ifElse';

describe('ifElse function', () => {
  const input = 1;
  const evenTransform = x => `${x} is even`;
  const oddTransform = x => `${x} is odd`;
  const isEven = x => 0 === x % 2;

  test('is a pure function', () => {
    ifElse(isEven, evenTransform, oddTransform)(input);
    expect(input).toMatchSnapshot();
  });

  test("1 |> ifElse(isEven, say it's even, say it's odd) === 1 is odd", () => {
    expect(ifElse(isEven, evenTransform, oddTransform)(1)).toMatchSnapshot();
  });

  test("2 |> ifElse(isEven, say it's even, say it's odd) === 2 is even", () => {
    expect(ifElse(isEven, evenTransform, oddTransform)(2)).toMatchSnapshot();
  });
});
