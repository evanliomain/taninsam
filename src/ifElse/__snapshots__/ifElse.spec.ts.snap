// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ifElse function 1 |> ifElse(isEven, say it's even, say it's odd) === 1 is odd 1`] = `"1 is odd"`;

exports[`ifElse function 2 |> ifElse(isEven, say it's even, say it's odd) === 2 is even 1`] = `"2 is even"`;

exports[`ifElse function is a pure function 1`] = `1`;
